admin_ui/backend/api/system.py

```
@@ -2759,6 +2772,82 @@ def _updates_status_cache_lock():
     return _UPDATES_STATUS_CACHE_LOCK
 
 
+_DOCKER_IMAGE_REF_RE = re.compile(r"^[a-z0-9]+(?:[._-][a-z0-9]+)*(?:/[a-z0-9]+(?:[._-][a-z0-9]+)*)*(?::[A-Za-z0-9][A-Za-z0-9._-]{0,127})?$")
+
+
+def _validate_docker_image_ref(ref: str) -> str:
+    r = (ref or "").strip()
+    if not r or r.startswith("-") or any(c.isspace() for c in r) or "\x00" in r:
+        raise ValueError("invalid docker image ref")
```

Ruff (0.14.14)

Avoid specifying long messages outside the exception class

(TRY003)